File: backend/src/main/java/com/novelwriting/entity/Character.java
package com.novelwriting.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "characters")
public class Character {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String personality;

    @Column(columnDefinition = "TEXT")
    private String appearance;

    @Column(columnDefinition = "TEXT")
    private String background;

    @Column(length = 50)
    private String role;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend-vue/src/types/character.ts
import type { ID, Timestamp } from './index'
import type { Tag } from './tag'

export interface Character {
  id: ID
  name: string
  role?: string
  description?: string
  personality?: string
  appearance?: string
  background?: string
  portraitImage?: string
  tags?: Tag[]
  createdAt?: Timestamp
  updatedAt?: Timestamp
}

export interface CharacterCreateDTO {
  name: string
  role?: string
  description?: string
  personality?: string
  appearance?: string
  background?: string
  portraitImage?: string
}

export interface CharacterUpdateDTO extends Partial<CharacterCreateDTO> {}
